executor/include/execution/trait/can_prefer.hpp

```diff
@@ -5,27 +5,37 @@
 #pragma once
 
 #include <execution/trait/can_require.hpp>
+#include <execution/trait/common_traits.hpp>
 #include <type_traits>
 
 namespace execution {
 
 template <typename Executor, typename Property,
           typename std::enable_if_t<
               Property::template is_applicable_property_v<Executor> &&
-                  Property::is_preferable,
+                  Property::is_preferable && can_require_v<Executor, Property>,
               int> = 0>
-Executor prefer(Executor ex, const Property t) {
-  if (execution::can_require_v<Executor, Property>) return ex.require(t);
-  return ex;
+constexpr auto prefer(Executor &&ex, const Property &p) noexcept {
+  return ex.require(p);
+}
+
+template <typename Executor, typename Property,
+          typename std::enable_if_t<
+              Property::template is_applicable_property_v<Executor> &&
+                  Property::is_preferable && !can_require_v<Executor, Property>,
+              int> = 0>
+constexpr auto prefer(Executor &&ex, const Property &p) noexcept {
+  return std::forward<Executor>(ex);
 }
 
 template <typename Executor, typename Properties, typename = void>
 struct can_prefer : std::false_type {};
 
 template <typename Executor, typename Property>
 struct can_prefer<Executor, Property,
-                  std::void_t<decltype(execution::require(
-                      std::declval<Executor>(), std::declval<Property>()))>>
+                  void_t<decltype(execution::prefer(
+                      std::declval<execution::remove_cv_ref_t<Executor>>(),
+                      std::declval<execution::remove_cv_ref_t<Property>>()))>>
```

Comment:
Are you required to remove the qualification here? Can you provide an example of case in which `can_prefer` was not working properly (when it's apparent that it should) and that forces you to remove qualification? 

I'm mentioning this because the function `prefer` is taking an universal reference for `Executor&&`, but for `can_prefer` to succeed, you force `prefer` to be called with normal `Executor` and `Property` types, when in theory it can support more types, with const qualifications and references.

Reply:
Your right, I am still not quite familiar with lvalue, rvalue references, etc.
I will read up more about them and get back. Definitely same places do not require to remove qualification and some required earlier and may not require now. Will review all of them and get back to you.

Reply:
This talk covers some important aspects 
https://www.youtube.com/watch?v=pIzaZbKUw2s

you might want to watch the first part as well, in case you feel that your move semantics understanding is shaky.